README.md

```diff
@@ -1,335 +1,446 @@
+7. [Development - Guide for contributing to the module](#development)
+
 ## Overview
 
-The apt module provides a simple interface for managing Apt source, key, and definitions with Puppet.
+The apt module lets you use Puppet to manage Apt sources, keys, and definitions.
 
 ## Module Description
 
-The apt module automates obtaining and installing software packages on \*nix systems.
-
-**Note**: While this module allows the use of short keys, **warnings are thrown if a full fingerprint is not used**, as they pose a serious security issue by opening you up to collision attacks.
+Apt (Advanced Package Tool) is a package manager available on Debian, Ubuntu, and several other operating systems. The apt module provides a series of classes, defines, types, and facts to help you automate Apt package management.
 
 ## Setup
 
-### What apt affects:
+### What apt affects
 
-* Package/service/configuration files for Apt
 * Your system's `sources.list` file and `sources.list.d` directory
 * System repositories
 * Authentication keys
 
+**Note:** Setting the apt module's `purge_sources_list` and `purge_sources_list_d` parameters to 'true' will destroy any existing content that you haven't declared through Puppet. The default for these parameters is 'false'.
+
 ### Beginning with apt
 
-To begin using the apt module with default parameters, declare the class with `include apt`.
+To use the apt module with default parameters, declare the `apt` class.
 
-Any Puppet code that uses anything from the apt module requires that the core apt class be declared.
+~~~
+include apt
+~~~
+
+**Note:** The main `apt` class is required by all other classes, types, and defines in this module. You must declare it whenever you use the module.
 
 ## Usage
 
-Using the apt module consists predominantly of declaring classes and defined types that provide the desired functionality and features. This module provides common resources and options that are shared by the various defined types in the apt module, so you **must always** include this class in your manifests.
+### Add GPG keys
+
+**Warning:** Using short key IDs presents a serious security issue, potentially leaving you open to collision attacks. We recommend you always use full fingerprints to identify your GPG keys. This module allows short keys, but issues a security warning if you use them.
+
+Declare the `apt::key` class:
+
+~~~
+apt::key { 'puppetlabs':
+  id      => '47B320EB4C7C375AA9DAE1A01054B7A24BD6EC30',
+  server  => 'pgp.mit.edu',
+  options => 'http-proxy="http://proxyuser:proxypass@example.org:3128"',
+}
+~~~
+
+You can make Apt load your key before others by adjusting the `priority` parameter (the default priority is 50).
+
+~~~
+apt::key { 'puppetlabs':
+  id       => '47B320EB4C7C375AA9DAE1A01054B7A24BD6EC30',
+  server   => 'pgp.mit.edu',
+  options  => 'http-proxy="http://proxyuser:proxypass@example.org:3128"',
+  priority => '50',
+}
+~~~
+
+### Prioritize backports
+
+~~~
```

Comment:
Docs style at this point is triple tildes for consistency and compatibility throughout our docs. Language-specific highlighting doesn't work on our main docs site.